### PR TITLE
fix: Improve randomness (#101)

### PR DESCRIPTION
**Pull request by hhartzer**
Fixes: #101

### CODE REVIEW

File: shortuuid/main.py [hunk 1]
```diff
@@ -104,8 +105,11 @@ def random(self, length: Optional[int] = None) -> str:
         if length is None:
             length = self._length
 
-        random_num = int(binascii.b2a_hex(os.urandom(length)), 16)
-        return int_to_string(random_num, self._alphabet, padding=length)[:length]
+        random_shortuuid = ""
+        while len(random_shortuuid) < length:
+            random_shortuuid += secrets.choice(self._alphabet)
+
+        return random_shortuuid
```
Comment (skorokithakis) [inline shortuuid/main.py:112]:
```suggestion
        return "".join(secrets.choice(self._alphabet) for _ in range(length))
```

**Review by skorokithakis — APPROVED** [review]

**Comment by hhartzer** [comment]
That's much more concise. Thank you for reviewing!

**Comment by skorokithakis** [comment]
Thanks for the code! Unfortunately I need to figure out ruff before I can merge, I screwed some stuff up.

**Comment by hhartzer** [comment]
No problem! Ruff is great and should be worth it in the end.

**Review by skorokithakis — APPROVED** [review]
Looks good, thanks!

**Pull request merged**